interop/src/server.rs

```diff
@@ -132,28 +151,37 @@ impl TestService for InteropTestService {
             sink.set_headers(metadata);
         }
         let f = async move {
-            while let Some(mut req) = stream.try_next().await? {
-                if req.has_response_status() {
-                    let code = req.get_response_status().get_code();
-                    let msg = req.take_response_status().take_message();
+            while let Some(req) = stream.try_next().await? {
+                if let Some(response_status) = &req.response_status.clone().into_option() {
+                    let code = response_status.code;
+                    let msg = String::from(&response_status.message);
                     let status = RpcStatus::with_message(code, msg);
                     sink.fail(status).await?;
                     return Ok(());
                 }
 
                 let mut resp = StreamingOutputCallResponse::default();
-                if let Some(param) = req.get_response_parameters().get(0) {
-                    resp.set_payload(util::new_payload(param.get_size() as usize));
+                if let Some(param) = req.response_parameters.get(0) {
+                    resp.payload = Some(util::new_payload(param.size as usize)).into();
                 }
                 // A workaround for timeout_on_sleeping_server test.
                 // The request only has 27182 bytes of zeros in payload.
                 //
                 // Client timeout 1ms is too short for grpcio. The server
                 // can response in 1ms. To make the test stable, the server
                 // sleeps 1s explicitly.
+
+                #[cfg(feature = "protobuf-codec")]
                 if req.get_payload().get_body().len() == 27182
```

Comment:
You can move this condition to last so that don't need to repeat all other conditions.

Reply:
Is this what you have in mind ?